wasm.cmake

```diff
@@ -259,6 +259,7 @@ set(CMAKE_C_FLAGS_RELWITHDEBINFO "-O2" CACHE STRING "Emscripten-overridden CMAKE
 set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG -O2" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_RELEASE")
 set(CMAKE_CXX_FLAGS_MINSIZEREL "-DNDEBUG -Os" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_MINSIZEREL")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2" CACHE STRING "Emscripten-overridden CMAKE_CXX_FLAGS_RELWITHDEBINFO")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -s USE_PTHREADS=1")
```

Comment:
I realize we need thems in the CMAKE_C_FLAGS too and  please add them in the CACHE as the other variable around. Toolchain file is parsed early, by adding variables to the cache you ensure they get treated correctly on future call to cmake.

```cmake
set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--shared-memory,--no-check-features --bind -s NO_EXIT_RUNTIME=1" CACHE STRING "Emscripten-overridden CMAKE_EXE_LINKER_FLAGS")
set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --bind -s NO_EXIT_RUNTIME=1" CACHE STRING "Emscripten-overridden")
```

Reply:
Additionally for sourcemaps to be generated in debug mode. We need to add -g4 to CMAKE_CXX_FLAGS_DEBUG.

Reply:
CMAKE_CXX_FLAGS_DEBUG doesn't exist. Does I create it ?

Reply:
Yes please for -g4